packages/tangle/messagefactory_test.go

```diff
@@ -146,38 +160,203 @@ func TestMessageFactory_POW(t *testing.T) {
+		likedInstead: map[ledgerstate.BranchID][]consensus.OpinionTuple{branches["3"]: {
+			consensus.OpinionTuple{Liked: branches["2"], Disliked: branches["3"]},
+			consensus.OpinionTuple{Liked: branches["1"], Disliked: branches["3"]},
+		}},
+	}
+	tangle.OTVConsensusManager = NewOTVConsensusManager(mockOTV)
+
+	references, err := PrepareLikeReferences(MessageIDs{testFramework.Message("3").ID(), testFramework.Message("2").ID()}, time.Now(), tangle)
+
+	require.NoError(t, err)
+
+	assert.Contains(t, references, testFramework.Message("1").ID())
+	assert.Contains(t, references, testFramework.Message("2").ID())
+	assert.Len(t, references, 2)
+}
+
+func TestMessageFactory_PrepareLikedReferences_2(t *testing.T) {
+	tangle := NewTestTangle()
+	defer tangle.Shutdown()
+
+	testFramework := NewMessageTestFramework(
+		tangle,
+		WithGenesisOutput("O1", 500),
+		WithGenesisOutput("O2", 500),
+	)
+
+	tangle.Setup()
+
+	tangle.Events.Error.Attach(events.NewClosure(func(err error) {
+		t.Logf("Error fired: %v", err)
+	}))
+
+	branches := make(map[string]ledgerstate.BranchID)
+
+	// Message 1
+	testFramework.CreateMessage("1", WithStrongParents("Genesis"), WithInputs("O1"), WithOutput("O3", 500), WithIssuingTime(time.Now().Add(5*time.Minute)))
+	testFramework.IssueMessages("1").WaitMessagesBooked()
+
+	// Message 2
+	testFramework.CreateMessage("2", WithStrongParents("Genesis"), WithInputs("O2"), WithOutput("O5", 500), WithIssuingTime(time.Now().Add(5*time.Minute)))
+	testFramework.IssueMessages("2").WaitMessagesBooked()
+
+	// Message 3
+	testFramework.CreateMessage("3", WithStrongParents("Genesis"), WithInputs("O2"), WithOutput("O4", 500))
+	testFramework.IssueMessages("3").WaitMessagesBooked()
+
+	// Message 4
+	testFramework.CreateMessage("4", WithStrongParents("Genesis"), WithInputs("O1"), WithOutput("O6", 500))
+	testFramework.IssueMessages("4").WaitMessagesBooked()
+
+	branches["1"], _ = transactionBranchID(tangle, testFramework.TransactionID("1"))
+	branches["2"], _ = transactionBranchID(tangle, testFramework.TransactionID("2"))
+	branches["3"], _ = transactionBranchID(tangle, testFramework.TransactionID("3"))
+	branches["4"], _ = transactionBranchID(tangle, testFramework.TransactionID("4"))
+
+	mockOTV := &SimpleMockOnTangleVoting{
+		disliked: ledgerstate.NewBranchIDs(branches["3"], branches["4"]),
+		likedInstead: map[ledgerstate.BranchID][]consensus.OpinionTuple{
+			branches["3"]: {consensus.OpinionTuple{Liked: branches["2"], Disliked: branches["3"]}},
+			branches["4"]: {consensus.OpinionTuple{Liked: branches["1"], Disliked: branches["4"]}},
+		},
+	}
+	tangle.OTVConsensusManager = NewOTVConsensusManager(mockOTV)
+
+	// Test first set of parents
+	references1, err := PrepareLikeReferences(MessageIDs{testFramework.Message("3").ID(), testFramework.Message("2").ID()}, time.Now(), tangle)
+	require.NoError(t, err)
+	assert.Contains(t, references1, testFramework.Message("2").ID())
+	assert.Len(t, references1, 1)
```

Comment:
This seems to be a good candidate to extract into a function that does the function call and all the checks for you. Might make the tests much more readable and shorter.

```
func checkReferences([all needed parameters], expected MessageIDs) {
    ref, err := PrepareLikeReferences()
    require.NoError(t, err)
    for e := range expected
        assert.Contains(t, references, e)
    }
    assert.Len(t, ref, len(expected))
}
```